packages/data-context/src/actions/ProjectActions.ts

```diff
@@ -458,7 +452,13 @@ export class ProjectActions {
 
     this.ctx.actions.project.setSpecs(specs)
 
-    return { specs, specPattern, excludeSpecPattern, additionalIgnorePattern }
+    this.ctx.project.startSpecWatcher(path, testingType, specPattern, excludeSpecPattern, additionalIgnorePattern)
+
+    if (testingType === 'component') {
+      this.api.getDevServer().updateSpecs(specs)
```

Comment:
Seems we have a pattern whenever we add/remove a spec:
* call `setSpecs()`
* call `devServer.updateSpecs()`
* trigger emitter

Should we move all this into `setSpecs`?

Reply:
Good call I moved this logic.